framework/permissions.py

```diff
@@ -142,10 +142,16 @@ def can_review_gate(
 
 
 def _maybe_redirect_to_login(handler_obj) -> flask.Response | dict:
-  # TODO(jrobbins): For API calls, we should return 401 rather than a redirect.
+  # Don't redirect if the handler is not an UI page (it is an API handler).
+  if not hasattr(handler_obj, 'get_common_data'):
+      return {}
```

Comment:
Thanks for taking care of this. I wasn't sure the best way to implement this, but this approach seems like a simple solution. 🙂